lib/menu_screen.dart

```diff
@@ -50,6 +50,9 @@ class _MenuScreenState extends State<MenuScreen> {
   }
 
   void runAnalysis() {
+    Navigator.pushNamed(
+      context,
+      LoadingScreen.id);
     if (!imageHandler.isMissingImages()) {
       // run analysis only if both images have been uploaded
       print("Run Analysis");
```

Comment:
The code to run the analysis should be inside the if statement, so that it only runs if both images are uploaded.

```suggestion
    if (!imageHandler.isMissingImages()) {
      // run analysis only if both images have been uploaded
      print("Run Analysis");
      Navigator.pushNamed(
        context,
        LoadingScreen.id);
```